tests/components/seventeentrack/__init__.py

```diff
@@ -1 +1,150 @@
+    "Undelivered": 1,
+    "Delivered": 1,
+    "Returned": 1,
+}
+
+VALID_CONFIG = {
+    CONF_USERNAME: "test",
+    CONF_PASSWORD: "test",
+}
+
+INVALID_CONFIG = {"notusername": "seventeentrack", "notpassword": "test"}
+
+VALID_OPTIONS = {
+    CONF_SHOW_ARCHIVED: True,
+    CONF_SHOW_DELIVERED: True,
+}
+
+NO_DELIVERED_OPTIONS = {
+    CONF_SHOW_ARCHIVED: False,
+    CONF_SHOW_DELIVERED: False,
+}
+
+
+async def _goto_future(hass, future=None):
+    """Move to future."""
+    if not future:
+        future = utcnow() + datetime.timedelta(minutes=10)
+    with patch("homeassistant.util.utcnow", return_value=future):
+        async_fire_time_changed(hass, future)
+        await hass.async_block_till_done()
+        async_fire_time_changed(hass, future)
+        await hass.async_block_till_done()
+
+
+class ProfileMock:
+    """ProfileMock will mock data coming from 17track."""
+
+    package_list = []
+    package_error = False
+    login_result = True
+    summary_data = DEFAULT_SUMMARY
+    summary_error = False
+    account_id = "123"
+
+    @classmethod
+    def reset(cls):
+        """Reset data to defaults."""
+        cls.package_list = []
+        cls.package_error = False
+        cls.login_result = True
+        cls.summary_data = DEFAULT_SUMMARY
+        cls.summary_error = False
+        cls.account_id = "123"
+
+    def __init__(self) -> None:
+        """Override Account id."""
+        self.account_id = self.__class__.account_id
+
+    async def login(self, email: str, password: str) -> bool:
+        """Login mock."""
+        if not self.__class__.login_result:
+            raise SeventeenTrackError("Login failed")
+        return self.__class__.login_result
+
+    async def packages(
+        self,
+        package_state: int | str = "",
+        show_archived: bool = False,
+        tz: str = "UTC",
+    ) -> list:
+        """Packages mock."""  # noqa: D401
+        if self.__class__.package_error:
+            raise SeventeenTrackError("Error")
+        return self.__class__.package_list[:]
+
+    async def summary(self, show_archived: bool = False) -> dict:
+        """Summary mock."""
+        if self.__class__.summary_error:
+            raise SeventeenTrackError("Error")
+        return self.__class__.summary_data
+
+
+class ClientMock:
+    """Mock the py17track client to inject the ProfileMock."""
+
+    def __init__(self, session) -> None:
+        """Mock the profile."""
+        self.profile = ProfileMock()
+
+
+async def init_integration(hass, config, options=None) -> MockConfigEntry:
+    """Set up the 17Track integration in Home Assistant."""
+
+    if options is None:
+        options = {}
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        title="17Track",
+        unique_id="0123456",
+        data=config,
+        options=options,
+    )
+
+    entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+
+    return entry
```

Comment:
Preferably we use normal mocks for these things (as in, an AsyncMock or MagicMock).

Tests I really recommend checking are `epion` or `analytics_insights`

Reply:
using AsyncMock now